### ISSUE TITLE
Issues with undocumented API detetction

### ISSUE THREAD

**Issue by guwirth** [SonarOpenCommunity/sonar-cxx#890]
Found some cases which are not properly detected and marked as "undocumented":

**Sample 1:**
```C++
/**
  * Sample bla bla
  */
    virtual AAA::BBB::IConfigurationContainer* AAA_CALL GetConfiguration() const noexcept override;
```

**Sample 2:**
```C++
/**
  * Implementation of @ref IStdTPConnection::CreateTxQueue().
  */
    virtual AAA::OLI::IError* OLI_CALL CreateTxQueue( // PRQA S 2121
        AAA::OLI::IStdTPTxQueue** ppoTxQueue) noexcept override;
```

**Sample 3:**
```C++
/**
  * Sample bla bla (empty line between comment & typedef)
  */

typedef enum OCI_CANBusParticipationMode
{

    /** 
     * Sample bla bla
     */
    OCI_BUSMODE_PASSIVE = 1,

} OCI_CANBusParticipationMode;
```

**Sample 4:**
```C++
/**
  * Sample bla bla
  */
typedef struct OCI_CANControllerCapabilities
{
  /** 
    * Sample bla bla
    */
    uint32 samplesPerBit;

} OCI_CANControllerCapabilities;

**Comment by guwirth** [comment]
partly solved by #969

**Comment by guwirth** [comment]
Closed with #976
